File: src/Model/Application/Budget.cs
namespace Model.Application
{
    public class Budget
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public bool IsActive { get; set; }

        public IEnumerable<BudgetPeriod> BudgetPeriods { get; set; }
        public IEnumerable<BudgetCategory> BudgetCategories { get; set; }
    }
}

File: src/budgetwallettsapp/src/types/api/budget.ts
import { BudgetPeriod } from "./budgetPeriod"
import { BudgetCategory } from "./budgetCategory"

export interface Budget {
    id: number,
    name: string,
    description: string,
    validFrom: Date,
    validTo: Date,
    isActive: boolean,
    budgetPeriods: BudgetPeriod[],
    budgetCategories: BudgetCategory[],
  }